clean.sh

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env bash
+
+usage()
+{
+    echo "Usage: clean [options]"
+    echo "Cleans the local dev environment."
+    echo
+    echo "  -b     Delete bin directory"
+    echo "  -p     Delete packages directory and NuGet package caches"
+    echo "  -a     Remove all untracked files in the working tree and delete NuGet package caches"
+    exit 1
+}
+
+# Parse arguments
+
+if [ $# == 0 ]
+then
+    echo "Please specify at least one option."
+    echo "Use 'clean -h' for help."
+    exit 1
+fi
+
+while [[ $# > 0 ]]
+do
+    opt="$1"
+    case $opt in
+        -h|--help)
+        usage
+        ;;
+        -b)
+        CleanBinDir=true
+        ;;
+        -p)
+        CleanPackagesDir=true
+        CleanPackagesCache=true
+        ;;
+        -a)
+        CleanWorkingTree=true
+        CleanPackagesCache=true
+        ;;
+        *)
+        echo "Unrecognized argument '$opt'"
+        echo "Use 'clean -h' for help."
+        exit 1
+        ;;
+    esac
+    shift
+done
+
+WorkingTreeRoot="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+if [ "$CleanBinDir" == true ] && [ "$CleanWorkingTree" != true ]
+then
+    echo "Removing bin directory"
+    rm -rf $WorkingTreeRoot/bin
+fi
+
+if [ "$CleanPackagesDir" == true ] && [ "$CleanWorkingTree" != true ]
+then
+    echo "Removing packages directory"
+    rm -rf $WorkingTreeRoot/packages
```

Comment:
That's a good point; I do see some value in putting the clean logic in MSBuild targets so that clean.cmd and clean.sh can both utilize it.  How many of these options do you think we should put in there, though?  I think ideally we would want a target corresponding to each option.  Does it make sense to invoke "git clean" from MSBuild, though?  Somehow that feels weird to me, but maybe it's fine.